test_unstructured/partition/test_email.py

```diff
@@ -54,565 +35,578 @@
-    with open(filename, "rb") as test_file:
-        spooled_temp_file = tempfile.SpooledTemporaryFile()
-        spooled_temp_file.write(test_file.read())
-        spooled_temp_file.seek(0)
-        elements = partition_email(file=spooled_temp_file)
-        assert len(elements) == 9
-        assert elements[3].text == "Make sure to RSVP!"
+def test_partition_email_partitions_a_text_plain_part_with_Base64_encoded_windows_1255_charset():
+    elements = partition_email(
+        example_doc_path("eml/email-no-utf8-2008-07-16.062410.eml"),
+        content_source="text/plain",
+    )
 
+    assert len(elements) == 30
+    assert elements[1].text.startswith("אני חושב שזה לא יהיה מקצועי והוגן שאני אראה לך היכן")
 
-def test_partition_email_from_text_file():
-    with open(example_doc_path("eml/fake-email.txt"), "rb") as f:
-        elements = partition_email(file=f, content_source="text/plain")
 
-    assert len(elements) > 0
-    assert elements == EXPECTED_OUTPUT
-    for element in elements:
-        assert element.metadata.filename is None
+def test_partition_email_partitions_an_html_part_with_quoted_printable_encoded_ISO_8859_1_charset():
+    elements = partition_email(
+        example_doc_path("eml/email-no-utf8-2014-03-17.111517.eml"),
+        content_source="text/html",
+        process_attachments=False,
+    )
 
+    assert len(elements) == 1
+    assert isinstance(elements[0], Table)
+    assert elements[0].text.startswith("Slava Gxyzxyz Hi Slava, The password for your Google")
 
-def test_partition_email_from_text_file_with_headers():
-    with open(example_doc_path("eml/fake-email.txt"), "rb") as f:
-        elements = partition_email(file=f, content_source="text/plain", include_headers=True)
 
-    assert len(elements) > 0
-    assert elements == ALL_EXPECTED_OUTPUT
-    for element in elements:
-        assert element.metadata.filename is None
+# -- edge-cases ----------------------------------------------------------------------------------
 
 
-def test_partition_email_from_text():
-    with open(example_doc_path("eml/fake-email.eml")) as f:
-        text = f.read()
+def test_partition_email_accepts_a_whitespace_only_file():
+    """Should produce no elements but should not raise an exception."""
+    assert partition_email(example_doc_path("eml/empty.eml")) == []
 
-    elements = partition_email(text=text)
 
-    assert len(elements) > 0
-    assert elements == EXPECTED_OUTPUT
-    for element in elements:
-        assert element.metadata.filename is None
+def test_partition_email_can_partition_an_empty_email():
+    assert (
+        partition_email(example_doc_path("eml/mime-no-body.eml"), process_attachments=False) == []
+    )
 
 
-def test_partition_email_from_text_work_with_empty_string():
-    assert partition_email(text="") == []
+def test_partition_email_does_not_break_on_an_encrypted_message():
+    assert (
+        partition_email(example_doc_path("eml/fake-encrypted.eml"), process_attachments=False) == []
+    )
```

Comment:
Is a warning logged, or does this fail silently? I believe it used to log a message.

Reply:
Well, it doesn't fail, it just produces no elements for the encrypted portions of the email.

Users don't have access to the logs so the only real option for communicating this is to fail with an exception. That would produce an HTTP 4xx and that could have a message. But failing is not appropriate here as far as I know.

Reply:
Don't open source users have access to the logs?